opencti-platform/opencti-graphql/src/database/stix-2-0-converter.ts

```diff
@@ -0,0 +1,83 @@
+import type { StoreEntity, StoreObject, StoreRelation } from '../types/store';
+import type * as S from '../types/stix-2-0-common';
+import type * as SDO from '../types/stix-2-0-sdo';
+import type * as SMO from '../types/stix-2-0-smo';
+import { INPUT_CREATED_BY, INPUT_EXTERNAL_REFS, INPUT_GRANTED_REFS, INPUT_KILLCHAIN, INPUT_LABELS, INPUT_MARKINGS } from '../schema/general';
+import { INPUT_OPERATING_SYSTEM, INPUT_SAMPLE } from '../schema/stixRefRelationship';
+import { ENTITY_TYPE_MALWARE } from '../schema/stixDomainObject';
+import { assertType, cleanObject, convertToStixDate } from './stix-converter-utils';
+
+// Meta
+const buildKillChainPhases = (instance: StoreEntity | StoreRelation): Array<SMO.StixInternalKillChainPhase> => {
+  return (instance[INPUT_KILLCHAIN] ?? []).map((k) => {
+    const data: SMO.StixInternalKillChainPhase = {
+      kill_chain_name: k.kill_chain_name,
+      phase_name: k.phase_name,
+      x_opencti_order: k.x_opencti_order,
+    };
+    return cleanObject(data);
+  });
+};
+
+const buildExternalReferences = (instance: StoreObject): Array<SMO.StixInternalExternalReference> => {
+  return (instance[INPUT_EXTERNAL_REFS] ?? []).map((e) => {
+    const data: SMO.StixInternalExternalReference = {
+      source_name: e.source_name,
+      description: e.description,
+      url: e.url,
+      hash: e.hashes, // key is "hash" in client python but should be "hashes"
+      external_id: e.external_id,
+    };
+    return cleanObject(data);
+  });
+};
+
+// Builders
+const buildStixObject = (instance: StoreObject): S.StixObject => {
+  return {
+    id: instance.standard_id,
+    x_opencti_id: instance.id,
+    spec_version: '2.0',
+    x_opencti_type: ENTITY_TYPE_MALWARE,
+    type: instance.entity_type.toLowerCase(),
+    x_opencti_granted_refs: (instance[INPUT_GRANTED_REFS] ?? []).map((m) => m.standard_id),
+    x_opencti_workflow_id: instance.x_opencti_workflow_id,
+  };
+};
+
+// General
+const buildStixDomain = (instance: StoreEntity | StoreRelation): S.StixDomainObject => {
+  return {
+    ...buildStixObject(instance),
+    created: convertToStixDate(instance.created),
+    modified: convertToStixDate(instance.modified),
+    revoked: instance.revoked,
+    confidence: instance.confidence,
+    // lang: instance.lang,
+    labels: (instance[INPUT_LABELS] ?? []).map((m) => m.value), // memo : x_opencti_labels for SCO
+    object_marking_refs: (instance[INPUT_MARKINGS] ?? []).map((m) => m.standard_id),
+    created_by_ref: instance[INPUT_CREATED_BY]?.standard_id, // memo: x_opencti_created_by_ref for SCO
+    external_references: buildExternalReferences(instance),
+  };
+};
+
+export const convertMalwareToStix2 = (instance: StoreEntity, type: string): SDO.StixMalware => {
```

Comment:
convertMalwareToStix could be a better name

Reply:
done